resources/netlog_defs.ts

```diff
@@ -26,8 +26,10 @@ export type LogDefinition = {
   };
   // map of indexes from a player id to the index of that player name
   playerIds?: { [fieldIdx: number]: number | null };
-  // a list of fields that are ok to not appear (or have invalid ids)
-  optionalFields?: readonly number[];
+  // a list of fields that are ok to be blank (or have invalid ids)
+  blankFields?: readonly number[];
+  // a list of fields that are ok to be completely missing including separator
+  optionalFields: readonly number[];
```

Comment:
Hmm, should this be "firstOptionalField"? Mostly in the sense that if one field is missing, then you can't have any fields after that one, right? (or if you do, then how do you differentiate [valid1, missing2, valid3] from [valid1, valid3]?

Reply:
The original implementation instead wrapped the entire field including the separator in an optional `(?:field)?` which allowed this, but it wouldn't work ultimately because that would match the fields you're describing.

The bigger issue here is that the property has to be specified on every definition or else it's not accessible without a specific hardcoded definition name, so what should be put in that property if there's no optional fields? And it can't be an optional property, TS won't detect the union that way. It could maybe be specified as `firstOptionalField: undefined`, or `null`, but it has to be present.